jscomp/frontend/ast_tdcls.ml

```diff
@@ -43,6 +43,11 @@ let newTdcls
              Parsetree.ptype_attributes = newAttrs}
          else x )
 
+let disable_unused_type: Parsetree.attribute =
+  { attr_name = Location.mknoloc "ocaml.warning";
+    attr_payload = PStr [ Str.eval (Exp.constant (Pconst_string ("-34", Location.none, None))) ];
```

Comment:
I think we should use the error name as this now works on 4.12 and will be less magic when debugging this on the future

```suggestion
    attr_payload = PStr [ Str.eval (Exp.constant (Pconst_string ("-unused-type-declaration", Location.none, None))) ];
```